docs/dev_doc.md

```diff
@@ -0,0 +1,197 @@
+
+The disambiguator should select one entity for each mention. It can use different strategies:
+
+- vector voter: base the entity selection on the vector similarities.
+- majority voter: base the entity selection on the different matching types votes.
+- priority voter: base the entity selection on the defined matching types priorities.
+- popularity voter: base the entity selection on the given entity weights.
+
+## Code
+
+### Coding style
+
+- [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html)
+
+- [Numpy docstrings style](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html)
+
+### Test
+
+Every development must be tested.
+The tests are launched with the `pytest` command.
+
+As we want to be an open source library, we must have a test coverage percentage close to 100%.
+The following command should be used to evaluate:
+
+```Bash
+pytest --cov-report term-missing --cov=. test
+```
+
+### Git best practices
+
+- Main branch must always be functional
+- When developing a new feature, a new branch named "feat/{feature_name}" must be created
+- When fixing an issue, a new branch named "fix/{feature_name}" must be created and linked to the issue
+- Create a merge/pull request to submit the code
+
+Start new developments
+
+```Bash
+git pull
+git checkout -b {new_branch}
+```
+
+Submit new developments
+
+```Bash
+git add {files}
+git commit -m "{comment}"
+git push {origin} {new_branch}
+```
+
+Amend commit and push after a review
+
+```Bash
+git checkout {branch_to_be_merged}
+git add {modified_files}
+git commit --amend
+git push --force-with-lease {origin} {branch_to_be_merged}
+```
+
+Update the code after an amend commit
+
+```
+git checkout {branch_to_be_merged}
+git fetch
+git reset --hard {origin}/{branch_to_be_merged}
+```
+
+### Virtual environment
+
+Setting up the virtual environment:
+
+- go to the project root directory: `buzz-el/`
+- create the virtual environment by running `virtualenv -p python3.10 {env/path}` (virtualenv needs to be installed) or `python3 -m venv {env/path}`. As a result, you should have a new folder at your project root.
+- activate the virtual environment by running
+  - `source {env/path}/bin/activate` on Linux
+  - `{env/path}\Scripts\activate` on Windows
+- check that the environment is properly installed by running the test from the project root directory with the command line `pytest test`.
+
+Setting up the workspace :
+
+- add the `src/` folder to the python paths by adding the full path `{path/to/the/project/}buzz-el/src` to the file:
+  - on Linux add `{path/to/the/project/}buzz-el/src` to the file `buzz-el/venv/lib/python3.10/site-packages/_virtualenv.pth`
+  - on Windows add `{C:\path\to\the\project\}buzz-el\src` to the file `{path/to/the/project/}buzz-el\venv\Lib\site-packages\_virtualenv.pth`
+- instead, the following command can also be run each time the project is used : `export PYTHONPATH="${PYTHONPATH}:{path/to/the/project/}buzz-el/src"`
```

Comment:
We should remove this part, however. It is useless since we now have defined a proper python package.

Reply:
Yes I agree